Add AggregateError to .any() in supported environments

JoschuaSchneider opened the pull request:
This adds the [AggregateError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/AggregateError/AggregateError) spec to the `SynchronousPromise.any` method. ([Promise.any](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/any#Description))

If `window` and `window.AggregateError` are defined, the method rejects with an instance of `AggregateError`.

I disabled JSHint for both of the blocks to allow the undefined `window` variable to be safely used after the checks.

The tests define a simple mock for the error and check if that implementation was used correctly.

This closes #24

Commit:
add support for AggregateError on .any() in browser

Commit:
add tests for AggregateError on .any() in supported environments

JoschuaSchneider commented:
I missed some semicolons in the tests, I'll fix those in a second

Commit:
add missing semicolons in test

fluffynuts commented on index.spec.js:
```diff
@@ -842,6 +842,43 @@ describe("synchronous-promise", function () {
       expect(capturedError).property("errors").to.contain("123");
     });
 
+    it("should reject with AggregateError if supported", function () {
```
minor: suggest putting this test in its own `describe` block and having the mocked `AggregateError` set up in a `beforeEach` and torn down in an `afterEach` -- covers the situation where the test, for some reason, fails.

JoschuaSchneider replied:
I'll give that a go, the only tiny problem I see is that this would never test the code without the `AggregateError` available.

JoschuaSchneider replied:
Nevermind, if I nest the beforeEach in the describe it is scoped to that test.

JoschuaSchneider replied:
I would call the test "static any (browsers with AggregateError)", do you have a better name for that?

Commit:
moved window mock into describe block with hooks and split up test

JoschuaSchneider commented:
I moved the test into a nested describe and put the mocking into hooks.

I also split up the test to make it more concice and descriptive, as the two cases affect different parts of the code.

fluffynuts commented:
My only request is to refactor out a function for the repeated logic of generating the correct Error type. Shouldn't take long. Please ping me when you've done so.

Commit:
move AggregateError creation into own function

fluffynuts reviewed (commented):
LGTM! Thanks for your contribution!
